### PR TITLE
chore(docs): update mockFunction API documentation

### LINKED ISSUE #81: Update mockFunction docs
# Summary

In https://github.com/sjbarag/brs/pull/622 we updated the `_brs_.mockFunction` API. The changes are backwards-compatible so the docs are still accurate, but let's update the docs to reflect the new features.

### PR DESCRIPTION
**Pull request by lkipke**
Closes #81

### CODE REVIEW

File: docs/api/reference/mocks.md [hunk 1]
```diff
@@ -42,18 +42,37 @@ Dynamically overwrite a function's implementation, and spy on the mocked impleme
...
 [Mock function](#mock-function-api)
 
 ### Usage 
+Original function definition:
 ```brightscript
-mock = _brs_.mockFunction("FunctionName", sub()
-    print "foobar"
-end sub)
+function myFunc(requiredArg as string)
+    return "myFunc: " requiredArg
```
Comment (sjbarag) [inline docs/api/reference/mocks.md:56]: I think you'll need a `+` between these values 😉 

```suggestion
    return "myFunc: " + requiredArg
```

File: docs/api/reference/mocks.md [hunk 2]
```diff
@@ -295,6 +314,31 @@ print mock.getMockName() ' => "fooBar"
 
 ------------
 
+## mock.mockReturnValue(value)
+Returns the name of the mocked function.
+
+### Parameters
+**funcName** `string` \
+The name of the function to overwrite. 
+------------
```
Comment (sjbarag) [inline docs/api/reference/mocks.md:323]: 🤔 these look like copy-pasta errors?

**Review by sjbarag — CHANGES_REQUESTED** [review]
Looks like a little copy-spaghetti spilled onto your keyboard here 😛 

Looks great once we get the docs for `mock.mockReturnValue(value)` fixed!

File: docs/api/reference/mocks.md (hunk 2 above)
Comment (lkipke) [reply docs/api/reference/mocks.md:323]: 🤦

File: docs/guides/testing-functions.md [hunk 3]
```diff
@@ -53,8 +53,8 @@ If `FooComponent` uses the `foo` function as a callback:
         <!-- Field with a callback we want to test -->
         <field id="myField" type="string" onChange="foo" />
 
-        <!-- Field with a callback we want to test -->
-        <field id="sideEffectField" type="string">
+        <!-- Field that gets modified as a side-effect of the `foo` function -->
```
Comment (lkipke) [inline docs/guides/testing-functions.md:56]: This fixes a different copy-lasagna error from #85... turns out when i write documentation i need to be a little more careful 😅

**Review by sjbarag — APPROVED** [review]
Looks like the copy-farfalle is all gone — thanks for the quick cleanup!

**Pull request merged**